src/routes/user.js

```diff
@@ -1,15 +1,13 @@
 import { Router } from 'express'
 import { create, getById, getAll, updateById } from '../controllers/user.js'
-import {
-  validateAuthentication,
-  validateTeacherRole
-} from '../middleware/auth.js'
+import { validateAuthentication } from '../middleware/auth.js'
+import { validateCanPatch } from '../utils/validationFunctions.js'
 
 const router = Router()
 
 router.post('/', create)
 router.get('/', validateAuthentication, getAll)
 router.get('/:id', validateAuthentication, getById)
-router.patch('/:id', validateAuthentication, validateTeacherRole, updateById)
+router.patch('/:id', validateAuthentication, validateCanPatch, updateById)
```

Comment:
let's not have `validateCanPatch` as a middleware, and instead call it in the controller